warehouse/encoding/parquetwriter.go

```diff
@@ -6,20 +6,21 @@ import (
 	"os"
 	"sort"
 
-	"github.com/rudderlabs/rudder-server/warehouse/internal/model"
-
-	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/xitongsys/parquet-go/writer"
 
 	"github.com/rudderlabs/rudder-server/utils/misc"
-	"github.com/xitongsys/parquet-go/writer"
+	"github.com/rudderlabs/rudder-server/warehouse/internal/model"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
 )
 
 const (
-	ParquetInt64           = "type=INT64, repetitiontype=OPTIONAL"
-	ParquetBoolean         = "type=BOOLEAN, repetitiontype=OPTIONAL"
-	ParquetDouble          = "type=DOUBLE, repetitiontype=OPTIONAL"
-	ParquetString          = "type=BYTE_ARRAY, convertedtype=UTF8, repetitiontype=OPTIONAL"
-	ParquetTimestampMicros = "type=INT64, convertedtype=TIMESTAMP_MICROS, repetitiontype=OPTIONAL"
+	ParquetInt64              = "type=INT64, repetitiontype=OPTIONAL"
+	ParquetBoolean            = "type=BOOLEAN, repetitiontype=OPTIONAL"
+	ParquetDouble             = "type=DOUBLE, repetitiontype=OPTIONAL"
+	ParquetString             = "type=BYTE_ARRAY, convertedtype=UTF8, repetitiontype=OPTIONAL"
+	ParquetTimestampMicrosNew = "type=INT64, logicaltype=TIMESTAMP, logicaltype.unit=MICROS, logicaltype.isadjustedtoutc=true, repetitiontype=OPTIONAL"
```

Comment:
Wouldn't adding `convertedtype=TIMESTAMP_MICROS` here help us having a single type being backwards & forwards compatible, instead of relying in two separate types?

Reply:
We can try but since we were planning to cut the release ASAP we just decided to go the safe way and have a flag, then fix it properly over time.